src/main/java/com/kodilla/ecommercee/UserController.java

```diff
@@ -0,0 +1,24 @@
+package com.kodilla.ecommercee;
+
+import com.kodilla.ecommercee.domain.UserDto;
+import org.springframework.web.bind.annotation.*;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@RestController
+@RequestMapping("/v1/user")
+public class UserController {
+
+    @RequestMapping(method = RequestMethod.POST, value = "createUser", consumes = APPLICATION_JSON_VALUE)
+    public void createUser(@RequestBody UserDto userDto) {
+    }
+
+    @RequestMapping(method = RequestMethod.PATCH, value = "blockUser")
+    public void blockUser(@RequestParam Long Id) {
+    }
+
+    @RequestMapping(method = RequestMethod.PATCH, value = "generateKey")
+    public void generateKey(@RequestParam Long Id) {
```

Comment:
Metoda powinna być `POST`

Reply:
Możesz podpowiedzieć dlaczego Paweł ? W tym przypadku rozumiem, że generuję tymczasowy klucz i częściowo zmieniam obiekt użytkownika przypisując wygenerowany klucz, czy w tym przypadku nie powinno być jednak PATCH ? Pewnie nie - bo tak napisales - a mógłbys podpowiedzieć dlaczego ?

Reply:
Czy ten klucz powinien być osobnym obiektem, który jest przypisany do Usera ?

Reply:
Metody POST używa się wtedy, gdy tworzone są nowe dane. Nazwa metody `generateKey` mówi, że tworzone będą nowe dane - generowany będzie klucz. To, gdzie i jak będzie on później zapisywany możemy obecnie pominąć - nie jest to widoczne z poziomu kontrolera.
Tak na marginesie: klucz może być zapisywany w obiekcie użytkownika ale może też być taka sytuacja, że każde generowanie klucza będzie tworzyć nowy rekord w tabeli